test-support/helpers/utility.js

```diff
@@ -0,0 +1,10 @@
+export function removeWhiteSpace(str) {
+  return str.replace(/\s+/g, ' ');
+}
+
+export function nodesToTextArray($nodeList) {
```

Comment:
This is an odd helper. Where have we used this before and for what purpose?

If we decide to keep it, I think we should add some yuidoc comments that explain how the function handles empty nodes, nested nodes, etc.

We would also be introducing some new nomenclature into the library with "node." I would prefer to use "element" or "tag" where appropriate, as these are the terms that are used by [MDN in their discussion of HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element). My assumption is that MDN pulls this from the HTML specs, so we would be well-served by using similar terminology.